WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorViewModel.kt

```diff
@@ -317,4 +337,39 @@ class ProductSelectorViewModel @Inject constructor(
     enum class LoadingState {
         IDLE, LOADING, APPENDING
     }
+
+    @Parcelize
+    sealed class SelectedItem(open val productId: Long) : Parcelable {
```

Comment:
Something that I realized while reading the code -- if you add base `SelectedItem::id` property, you won't need the helper `productAndVariationIds`, `variationIds`, or `productIds` properties. You'll always have the right ones using a simple `map()` call:
```suggestion
    sealed class SelectedItem(val id: Long) : Parcelable {
        @Parcelize
        data class ProductOrVariation(val productOrVariationId: Long) : SelectedItem(productOrVariationId)
        @Parcelize
        data class Product(val productId: Long) : SelectedItem(productId)
        @Parcelize
        data class ProductVariation(val productId: Long, val variationId: Long) : SelectedItem(variationId)
    }
```

Reply:
That's a neat idea, thanks! Done with 929b743eefc049b7549cc21a58c22d79a67953a8.